src/components/label/label.tsx

```diff
@@ -0,0 +1,87 @@
+import { getIconName } from '../icon/get-icon-props';
+import { Component, Prop, h } from '@stencil/core';
+import { Label, LabelValue } from './label.types';
+import { Icon } from '../../interface';
+
+/**
+ * This components displays a different label depending on the current given
+ * value. A label can consist of a text and an optional icon. If no matching label is
+ * found, the default label will be displayed.
+ *
+ * One use case of the component is to enhance the visualization of a `boolean`
+ * field like a checkbox or switch in a `readonly` state.
+ *
+ * The reason we offer this component is that the default styling
+ * of the Checkbox or Toggle switch in the `readonly` state may not always
+ * provide the best way of _visualizing information_, potentially leading to
+ * confusion and negatively affecting the end-users' experience.
+ *
+ * @exampleComponent limel-example-label
+ * @beta
+ */
+@Component({
+    tag: 'limel-label',
+    shadow: true,
+    styleUrl: 'label.scss',
+})
+export class DynamicLabel {
+    /**
+     * The value of the component.
+     *
+     * The value will be matched with the given labels to determine what label
+     * to display
+     */
+    @Prop({ reflect: true })
+    public value: LabelValue;
+
+    /**
+     * Default label to display if no label with corresponding value is found
+     */
+    @Prop({ reflect: true })
+    public defaultLabel: Omit<Label, 'value'> = {};
+
+    /**
+     * Available labels
+     */
+    @Prop()
+    public labels: Label[] = [];
```

Comment:
I can't say that I read these docs and understand what these props do 😅 
Even when I look at the code, I am still confused. But the consumer only sees the props' description in the generated docs. So the docs should make it clear what these props do.

I tried to improve the docs like below:

```suggestion
    /**
     * The current value of the component; 
     * which is used to match with the `labels` array to determine the label to display. 
     * If no matching label is found, the `defaultLabel` is displayed.
     */
    @Prop({ reflect: true })
    public value: LabelValue;

    /**
     * The label to display when no matching value is found in the `labels` array. 
     * This is a fallback label that ensures there's always a label displayed for the component.
     */
    @Prop({ reflect: true })
    public defaultLabel: Omit<Label, 'value'> = {};

    /**
     * An array of available labels. Each label in this array has a corresponding value. 
     * When the `value` prop matches a value in this array, the corresponding label is displayed.
     */
    @Prop()
    public labels: Label[] = [];
```

However, I am still unsatisfied and confused 😕 .

What if we rename the component to `limel-dynamic-label` and then we can rename the props like these:


```suggestion
    /**
     * The current value of the component; 
     * which is used to match with the `dynamicLabels` array to determine the label to display. 
     * If no matching label is found, the `label` is displayed.
     */
    @Prop({ reflect: true })
    public value: LabelValue;

    /**
     * The label to display when no matching value is found in the `dynamicLabels` array. 
     * This is a fallback label that ensures there's always a label displayed for the component.
     */
    @Prop({ reflect: true })
    public label: Omit<Label, 'value'> = {};

    /**
     * An array of available labels which can be dynamically applied based on the given value. 
     * Each label in this array will have to have a corresponding value. 
     * When the `value` prop matches a value in this array, 
     * the corresponding label is dynamically displayed.
     */
    @Prop()
    public dynamicLabels: Label[] = [];
```

A bit better. But what do you think?

Reply:
I updated the descriptions but did not change the names

bc895e20bdc592e497d6de86744a4aaf7be96be0

Reply:
I'm OK if you wanna squash all these and merge them into my branch. I think we can continue from there more easily.

Reply:
We can't squash them here easily, lets just merge it to the other branch first

Reply:
sure. I just mean things are OK from my perspective